### ISSUE TITLE
Copy task expand() transforms "\n" into newlines

### ISSUE THREAD

**Issue by cdalexndr** [gradle/gradle#10458]
Consider the input some configuration file that contains the following text in some json:
`"SELECT\n  count(id) AS idc\nFROM dev.product"`

Apply the following copy task with expand:
```
task copyTest(type: Copy) {
    from 'test'
    into "$buildDir/test"
    expand()
}
```

The text is transformed into an invalid json:
```
"SELECT
  count(id) AS idc
FROM dev.product"
```

------------------------------------------------------------
Gradle 5.5.1
------------------------------------------------------------

Build time:   2019-07-10 20:38:12 UTC
Revision:     3245f748c7061472da4dc184991919810f7935a5

Kotlin:       1.3.31
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.14 compiled on March 12 2019
JVM:          12.0.2 (Oracle Corporation 12.0.2+10)
OS:           Windows 10 10.0 amd64

**Comment by cdalexndr** [comment]
Workaround: escaped "\\\n" string instead of "\n"

**Comment by cdalexndr** [comment]
Same behaviour with "\t"

**Comment by big-guy** [comment]
`expand()` makes Gradle treat every file copied as if it's a template.  It's run through Groovy's [SimpleTemplateEngine](http://docs.groovy-lang.org/docs/next/html/documentation/template-engines.html#_simpletemplateengine).  

If you don't want this behavior for some files, you can exclude them and copy them separately or escape the files (like you found).

**Comment by cdalexndr** [comment]
SimpleTemplateEngine has an option to escape backslashes:
[SimpleTemplateEngine.setEscapeBackslash​(boolean escapeBackslash)](http://docs.groovy-lang.org/latest/html/api/groovy/text/SimpleTemplateEngine.html#setEscapeBackslash(boolean))
Turning on this option fixes my issue: [online script](https://groovyconsole.appspot.com/edit/5074803962675200)
Please provide a method to turn on this option when using [expand](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/AbstractCopyTask.html#expand-java.util.Map-)

**Comment by cdalexndr** [comment]
Opened feature request: https://github.com/gradle/gradle/issues/10492

**Comment by ljacomet** [comment]
While this was closed as "won't fix", the linked issue (#10492) ended up being fixed in 7.2
